inventory.js

```diff
@@ -1,67 +1,33 @@
-function Inventory(inventory)
+function Inventory(inventory={})
 {
-    if (inventory === undefined) {
-        //Data Members:
-        this.list = [[""]];
-    } else {
-        // Restore from JSON
-        this.list = inventory.list;
-    }
+    this.inventory = inventory;
 
-    //Member Functions:
-    this.add_item = function(new_item)
-    {
-        var list_size = this.list.length;
-        var already_exists = false;
-        for(var i = 0; i < this.list.length; i++)
-        {
-            if(this.list[i][0] == new_item)
-            {
-                //alert("This item already exists in your inventory");
-                already_exists = true;
-                //Add the item to the array that already has another instance of the same item.
-                this.list[i].push(new_item);
-            }
-        }
-        //If the item doesn't exist anywhere in the list, we should push a new array onto the list.
-        if(!already_exists)
-        {
-            //alert("This item did not exist in your inventory, so we will add it.");
-            this.list.push([new_item]);
-        }
+    this.add_item = function(item){
+      if(item in this.inventory)
+        this.inventory[item] += 1;
+      else
+        this.inventory[item] = 1;
     }
+
+    this.costs = {
+      "Rock": 1,
+      "Shears": 35,
+      "Axe": 30,
+      "Hatchet": 15,
+      "Chainsaw": 60,
+      "Chisel": 5,
+      "Sledge": 25,
+      "Jackhammer": 100,
+      "Machete": 25,
+      "Binoculars": 50
+    };
+
     this.display_inventory = function()
     {
-        var inventory_to_html = "<h3>Inventory</h3>";
-        //Code for placing inventory items in the string goes here
-        //go through the array of arrays.  At each subarray, add that subarray[0] to the html, and add "x" and add the subarray's size.
-        for(var i = 1; i < this.list.length; i++)
-        {
-            var cost = 0;
-            if(this.list[i][0] == "Rock")
-                cost = 1;
-            else if(this.list[i][0] == "Shears")
-                cost = 35;
-            else if(this.list[i][0] == "Axe")
-                cost = 30;
-            else if(this.list[i][0] == "Hatchet")
-                cost = 15;
-            else if(this.list[i][0] == "Chainsaw")
-                cost = 60;
-            else if(this.list[i][0] == "Chisel")
-                cost = 5;
-            else if(this.list[i][0] == "Sledge")
-                cost = 25;
-            else if(this.list[i][0] == "Jackhammer")
-                cost = 100;
-            else if(this.list[i][0] == "Machete")
-                cost = 25;
-            else if(this.list[i][0] == "Binoculars")
-                cost = 50;
-            else
-                cost = 1;
-            inventory_to_html += ("<p>" + this.list[i][0] + " x" + this.list[i].length + " ---- Cost: $" + cost + " ea.</p>");
+        let inventory_to_html = "<h3>Inventory</h3>";
+        for (let [item, count] of Object.entries(this.inventory)) {
```

Comment:
This is a nifty way of for-looping.  Is this a commonly used syntax?  It's is certainly very succinct, but I do find it just a tad bit hard to read quickly/clearly. 

Is there perhaps a more verbose way of expressing this that is not too DRY?  Like maybe saying something like:
`for(i in inventory){
   let item = inventory.item
   let count = inventory.count`

Reply:
I've changed it to using a forEach function for familiarity